### DISCUSSION TITLE
We can execute .net application!?

### DISCUSSION THREAD

**Discussion by sajjadarashhh** [CosmosOS/Cosmos#2261] [Ideas]
We can execute .net maui applications in cosmos os?

**Comment by valentinbreiz** [comment]
No

**Comment by sajjadarashhh** [comment]
@valentinbreiz How can i run .net applications in cosmos!? or write applications for this platform

**Reply by Samma2009** [reply]
yes

**Comment by BL1NDX3N0N** [comment]
Out of the box? No. Theoretically? Yes.

First you need to know that .NET is a runtime and you would need to port the runtime to run on your OS. Second of all, that runtime makes a lot of calls into OS specific APIs, those APIs would also need to be implemented inside of your OS. This would be a massive endeavor for a single individual.

**Reply by MishaTy** [reply]
It is possible and I wrote a .net runtime. Join my discord server to suggest features to be added.
http://github.com/MishaTY/Dotnetparser

**Reply by retrievaaaa** [reply]
@MishaTY that sounds interesting for running external apps from files, which C# version does it support? Does it depend on Reflection?

**Reply by MishaTy** [reply]
It should be able to load any c# version, and it does not depend on reflection.
